examples/zephyr/dfu/README.md

```diff
@@ -0,0 +1,273 @@
+# Golioth DFU sample
+
+## Overview
+
+This sample application demonstrates how to perform Device Firmware
+Upgrade (DFU) using Golioth's Over-the-Air (OTA) update service. This is
+a two step process:
+
+* Build initial firmware and flash it to the device.
+* Build new firmware to use as the upgrade and upload it to Golioth.
+
+Your device automatically detects available releases from Golioth and
+performs an OTA update when a newer version is available.
+
+## Requirements
+
+* Golioth credentials
+* Network connectivity
+
+## Building and Running
+
+### Authentication specific configuration
+
+#### PSK based auth
+
+Configure the following Kconfig options based on your Golioth
+credentials:
+
+* GOLIOTH_SAMPLE_PSK_ID - PSK ID of registered device
+* GOLIOTH_SAMPLE_PSK - PSK of registered device
+
+by adding these lines to configuration file (e.g. `prj.conf`):
+
+```cfg
+CONFIG_GOLIOTH_SAMPLE_PSK_ID="my-psk-id"
+CONFIG_GOLIOTH_SAMPLE_PSK="my-psk"
+```
+
+#### Certificate based auth
+
+Configure the following Kconfig options based on your Golioth
+credentials:
+
+* CONFIG_GOLIOTH_AUTH_METHOD_CERT - use certificate-based
+    authentication
+* CONFIG_GOLIOTH_SAMPLE_HARDCODED_CRT_PATH - device certificate
+* CONFIG_GOLIOTH_SAMPLE_HARDCODED_KEY_PATH - device private key
+
+by adding these lines to configuration file (e.g. `prj.conf`):
+
+```cfg
+CONFIG_GOLIOTH_AUTH_METHOD_CERT=y
+CONFIG_GOLIOTH_SAMPLE_HARDCODED_CRT_PATH="keys/device.crt.der"
+CONFIG_GOLIOTH_SAMPLE_HARDCODED_KEY_PATH="keys/device.key.der"
+```
+
+### Platform specific configuration
+
+#### nRF52840 DK + ESP32-WROOM-32
+
+This subsection documents using nRF52840 DK running Zephyr with
+offloaded ESP-AT WiFi driver and ESP32-WROOM-32 module based board (such
+as ESP32 DevkitC rev. 4) running WiFi stack. See [AT Binary
+Lists](https://docs.espressif.com/projects/esp-at/en/latest/AT_Binary_Lists/index.html)
+for links to ESP-AT binaries and details on how to flash ESP-AT image on
+ESP chip. Flash ESP chip with following command:
+
+```console
+esptool.py write_flash --verify 0x0 PATH_TO_ESP_AT/factory/factory_WROOM-32.bin
+```
+
+Connect nRF52840 DK and ESP32-DevKitC V4 (or other
+ESP32-WROOM-32/ESP32-ROVER-32 based board) using wires:
+
+  | nRF52840 DK | ESP32-WROOM-32  | ESP32-WROVER-32 |
+  | ----------- | --------------- | ----------------|
+  | P1.01 (RX)  | IO17 (TX)       | IO22 (TX)       |
+  | P1.02 (TX)  | IO16 (RX)       | IO19 (RX)       |
+  | P1.03 (CTS) | IO14 (RTS)      | IO14 (RTS)      |
+  | P1.04 (RTS) | IO15 (CTS)      | IO15 (CTS)      |
+  | P1.05       | EN              | EN              |
+  | GND         | GND             | GND             |
```

Comment:
Note that I added pinouts for the WROVER version of the ESP32. This is what I use and I had to look up the pinout to connect my test unit.